lib/matplotlib/backend_bases.py

```diff
@@ -3195,3 +3207,560 @@ def zoom(self, *args):
+            - 'tool_added_event'
+
+            For every tool added a new event is created
+
+            - 'tool_trigger_TOOLNAME`
+              Where TOOLNAME is the id of the tool.
+
+        func : function
+            Function to be called with signature
+            def func(event)
+        """
+        return self._callbacks.connect(s, func)
+
+    def nav_disconnect(self, cid):
+        """
+        Disconnect callback id cid
+
+        Example usage::
+
+            cid = navigation.nav_connect('tool_trigger_zoom', on_press)
+            #...later
+            navigation.nav_disconnect(cid)
+        """
+        return self._callbacks.disconnect(cid)
+
+    def message_event(self, message, sender=None):
+        """ Emit a `NavigationMessageEvent`"""
+        if sender is None:
+            sender = self
+
+        s = 'tool_message_event'
+        event = NavigationMessageEvent(s, sender, message)
+        self._callbacks.process(s, event)
+
+    @property
+    def active_toggle(self):
+        """
+        Toggled Tool
+
+        **dict** :  Currently toggled tools
+        """
+
+        return self._toggled
+
+    def get_tool_keymap(self, name):
+        """
+        Get the keymap associated with the specified tool
+
+        Parameters
+        ----------
+        name : string
+            Name of the Tool
+
+        Returns
+        ----------
+        list : list of keys associated with the Tool
+        """
+
+        keys = [k for k, i in six.iteritems(self._keys) if i == name]
+        return keys
+
+    def _remove_keys(self, name):
+        for k in self.get_tool_keymap(name):
+            del self._keys[k]
+
+    def set_tool_keymap(self, name, *keys):
+        """
+        Set the keymap to associate with the specified tool
+
+        Parameters
+        ----------
+        name : string
+            Name of the Tool
+        keys : keys to associate with the Tool
+        """
+
+        if name not in self._tools:
+            raise KeyError('%s not in Tools' % name)
+
+        self._remove_keys(name)
+
+        for key in keys:
+            for k in validate_stringlist(key):
+                if k in self._keys:
+                    warnings.warn('Key %s changed from %s to %s' %
+                                  (k, self._keys[k], name))
+                self._keys[k] = name
+
+    def remove_tool(self, name):
+        """
+        Remove tool from `Navigation`
+
+        Parameters
+        ----------
+        name : string
+            Name of the Tool
+        """
+
+        tool = self.get_tool(name)
+        tool.destroy()
+
+        # If is a toggle tool and toggled, untoggle
+        if getattr(tool, 'toggled', False):
+            self.tool_trigger_event(tool, 'navigation')
+
+        self._remove_keys(name)
+
+        s = 'tool_removed_event'
+        event = ToolEvent(s, self, tool)
+        self._callbacks.process(s, event)
+
+        del self._tools[name]
+
+    def add_tools(self, tools):
```

Comment:
great, sold on :-1: for args magic (we have it else where where I do not like it, but supposedly users like that sort of thing)

What is the use case for bulk adding tools?  Writing this loop in user code does not strike me as super onerous.

My worry with the typo is getting an exception about args being wrong (which leads you to check what you are passing in) rather than an `AttributeError` (which leads you to check your spelling).